### PR TITLE
STYLE: Fix miscellaneous Python warnings

### PR DESCRIPTION
**Pull request by jhlegarreta**
- STYLE: Fix dictionary literal creation warnings
- STYLE: Fix list literal creation warnings
- STYLE: Fix no-effect statements
- STYLE: Compare to `None` with `is`/`is not`
- STYLE: Remove redundant parentheses

### CODE REVIEW

**Comment by jhlegarreta** [comment]
I'd say that the reported failure is unrelated to this patch set:
https://github.com/dipy/dipy/runs/6337157843?check_suite_focus=true#step:9:3753

The parallelization builds seem to be failing on and off for different tests.

**Comment by jhlegarreta** [comment]
Again, failures are not related this time either:
https://github.com/dipy/dipy/runs/6341748131?check_suite_focus=true#step:9:3988

This has been failing on and off as well, e.g. https://github.com/dipy/dipy/pull/2533#issuecomment-1047217940

File: dipy/workflows/viz.py [hunk 1]
```diff
@@ -115,7 +115,7 @@ def run(self, input_files, cluster=False, cluster_thr=15.,
         world_coords = not native_coords
         bundle_colors = None
 
-        mni_2009a = {}
+        mni_2009a = dict({})
```
Comment (skoudoro) [inline dipy/workflows/viz.py:118]: same comment as above

File: dipy/workflows/viz.py [hunk 2]
```diff
@@ -124,7 +124,7 @@ def run(self, input_files, cluster=False, cluster_thr=15.,
         mni_2009a['vox_size'] = (1., 1., 1.)
         mni_2009a['vox_space'] = 'RAS'
 
-        mni_2009c = {}
+        mni_2009c = dict({})
```
Comment (skoudoro) [inline dipy/workflows/viz.py:127]: same comment as above

File: doc/examples/histology_resdnn.py [hunk 3]
```diff
@@ -107,7 +107,7 @@
 of the corpus callosum and a single-fiber population should be visible.
 """
 
-camera = {}
+camera = dict({})
 camera['zoom_factor'] = 0.85
 camera['view_position'] = np.array([(ori_shape[0] - 1) / 2.0,
```
Comment (skoudoro) [inline doc/examples/histology_resdnn.py:112]: you can initialize it like you did above (meaning, ` camera = {'zoom_factor': 0.85 ....}`)

File: doc/examples/reconst_gqi.py [hunk 4]
```diff
@@ -127,7 +127,7 @@
 the same data and mask are used.
 """
 
-np.sum(gqpeaks.odf != ODF) == 0
+print(np.sum(gqpeaks.odf != ODF) == 0)
```
Comment (skoudoro) [inline doc/examples/reconst_gqi.py:130]: I prefer that you remove the comment below and remove the `print`. Not sure this information is valuable.

**Review by skoudoro — COMMENTED** [review]
Hi @jhlegarreta,

Always good to see some code cleaning! Thank you for that. 

See below for some comments. After that, it should be ready to go.

File: doc/examples/reconst_gqi.py (hunk 4 above)
Comment (jhlegarreta) [reply doc/examples/reconst_gqi.py:130]: I think it is valuable as it proofs what is said in the immediately previous comment:
```
This ODF will be of course identical to the ODF calculated above as long as
the same data and mask are used.
```

So I'd keep it.

File: dipy/align/imaffine.py [hunk 5]
```diff
@@ -61,7 +61,7 @@
 from dipy.utils.deprecator import deprecated_params
 
 _interp_options = ['nearest', 'linear']
-_transform_method = {}
+_transform_method = dict({})
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply dipy/align/imaffine.py]: Done in 5517829.

File: dipy/align/transforms.pyx [hunk 6]
```diff
@@ -1463,7 +1463,7 @@ cdef class AffineTransform3D(Transform):
         R[3, 3] = 1
 
 
-regtransforms = {}
+regtransforms = dict({})
```
[the remark this answers is not in the document]
Comment (jhlegarreta) [reply dipy/align/transforms.pyx]: Done in 5517829.

File: dipy/workflows/viz.py (hunk 1 above)
Comment (jhlegarreta) [reply dipy/workflows/viz.py]: Done in 5517829.

File: dipy/workflows/viz.py (hunk 2 above)
Comment (jhlegarreta) [reply dipy/workflows/viz.py]: Done in 5517829.

File: doc/examples/histology_resdnn.py (hunk 3 above)
Comment (jhlegarreta) [reply doc/examples/histology_resdnn.py]: Done in 5517829.

File: doc/examples/reconst_gqi.py (hunk 4 above)
Comment (skoudoro) [reply doc/examples/reconst_gqi.py:130]: ok, that's fine.

**Comment by jhlegarreta** [comment]
Unrelated failure in parallelization build, cross-ref https://github.com/dipy/dipy/pull/2592#issuecomment-1120325684

**Comment by skoudoro** [comment]
Thanks, all good! merging

**Pull request merged**